components/sup/src/manager/mod.rs

```diff
@@ -1549,7 +1554,15 @@ impl Manager {
         if !is_first {
             writer.write(",".as_bytes())?;
         }
-        serde_json::to_writer(writer, service)
+
+        let mut value = serde_json::to_value(service).unwrap();
+        {
+            let service_value = value.as_object_mut().unwrap();
+            if self.hide_secrets {
+                service_value.remove("cfg");
+            }
+        }
+        serde_json::to_writer(writer, &value)
```

Comment:
This works as a fast way to change existing behavior, but in the long run, I'd rather take a more secure-by-default approach where we change the implementation of the serialize trait rather than pull things out after the fact. If a different code path started serializing `Service`, this wouldn't apply.

Also, defaulting to include rather than exclude is more likely to lead to unintended exposure in the future. We should explicitly enumerate the fields we *do* want rather than the ones we want to omit. An explicit implementation of `Serialize` would achieve that as well.

In any case, it doesn't have to be now if we're pressed for time, but at least a TODO here and an issue to track it would be good.